scripts/background-tasks.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+# Acquia hosting site / environment names
+site="$1"
+env="$2"
+# database role. (Not expected to be needed in most hook scripts.)
+db_role="$3"
+# The public domain name of the website.
+uri="$4"
+
+# The websites' document root
+docroot="/var/www/html/$site.$env/docroot"
+
+# Create and set Drush cache to unique local temporary storage per site.
+# This approach isolates drush processes to completely avoid race conditions
+# that persist after initial attempts at addressing in BLT: https://github.com/acquia/blt/pull/2922
+cacheDir=`/usr/bin/env php /mnt/www/html/$site.$env/vendor/acquia/blt/scripts/blt/drush/cache.php $site $env $uri`
+# Print to cloud task log.
+echo "Generated temporary drush cache directory: $cacheDir."
+
+DRUSH_CMD="drush9 --root=$docroot --uri=$uri"
+
+function set_maintenance_mode {
+  echo "Setting site maintenance_mode to $1."
+
+  DRUSH_PATHS_CACHE_DIRECTORY=$cacheDir $DRUSH_CMD sset system.maintenance_mode $1 -y
+  maintenance_mode="$(DRUSH_PATHS_CACHE_DIRECTORY=$cacheDir $DRUSH_CMD sget system.maintenance_mode)"
+  echo "Site maintenance mode: $maintenance_mode"
+}
+
+function set_site_owner_maintenance_mode {
+  echo "Setting site_owner_maintenance_mode to $1."
+
+  DRUSH_PATHS_CACHE_DIRECTORY=$cacheDir $DRUSH_CMD cset acsf.settings site_owner_maintenance_mode $1 -y
+  site_owner_maintenance_mode="$(DRUSH_PATHS_CACHE_DIRECTORY=$cacheDir $DRUSH_CMD cget acsf.settings site_owner_maintenance_mode -y)"
+  echo "Site owner maintenance mode: $site_owner_maintenance_mode"
+}
+
+set_site_owner_maintenance_mode 0
+
+# Fetch if import is pending.
+CMD="$(DRUSH_PATHS_CACHE_DIRECTORY=$cacheDir $DRUSH_CMD sget cohesion_import_pending  --format=json | grep cohesion_import_pending | grep -o '.$')"
```

Comment:
Why does this mention cohesion?

Reply:
This is an example use case. I think it works well for illustration purposes.